api/k8s/v1beta4/k8s_helpers.go

```diff
@@ -47,3 +47,60 @@ func (s *API) WaitForCluster(req *WaitForClusterRequest) (*Cluster, error) {
 	}
 	return cluster.(*Cluster), nil
 }
+
+func (s *API) getNodesWhenReady(req *ListNodesRequest, expectedCount uint32) (interface{}, bool, error) {
+	res, err := s.ListNodes(req, scw.WithAllPages())
+	if err != nil {
+		return nil, false, err
+	}
+
+	// first check that we have the right nodes count
+	if res.TotalCount != expectedCount {
+		return nil, false, nil
+	}
+
+	// then check that each node has Ready status
+	for _, node := range res.Nodes {
+		if node.Status != NodeStatusReady {
+			return nil, false, nil
+		}
+	}
+
+	return res, true, nil
+}
+
+// WaitForPoolNodesReadyRequest is used by WaitForPoolNodesReady method.
+type WaitForPoolNodesReadyRequest struct {
+	PoolID  string
+	Timeout time.Duration
+}
+
+// WaitForPoolNodesReady waits for the nodes of a pool to be ready
+func (s *API) WaitForPoolNodesReady(req *WaitForPoolNodesReadyRequest) error {
+	pool, err := s.GetPool(&GetPoolRequest{
+		PoolID: req.PoolID,
+	})
+	if err != nil {
+		return err
+	}
+
+	r := &ListNodesRequest{
```

Comment:
You don't need to list the nodes, you have `pool.CurrentNodeCount` :)

Reply:
No more nodes listing after API fix